core/src/main/java/org/hibernate/ogm/loader/impl/OgmLoader.java

```diff
@@ -360,6 +419,10 @@ else if ( ogmLoadingContext.hasResultSet() ) {
 		return results;
 	}
 
+	private boolean useBatching(QueryParameters qp) {
+		return qp.getPositionalParameterValues().length > 1;
```

Comment:
Ok, got it. The method name made me stumble, I thought it's about whether batching is enabled at all or not. Maybe `containsSeveralIds()` or so?